java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,431 @@
+    }
+
+    private static void updateVector(BitVector bitVector, boolean value, int rowCount) {
+        bitVector.setSafe(rowCount, value? 1: 0);
+        bitVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(TinyIntVector tinyIntVector, int value, int rowCount) {
+        tinyIntVector.setSafe(rowCount, value);
+        tinyIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(SmallIntVector smallIntVector, int value, int rowCount) {
+        smallIntVector.setSafe(rowCount, value);
+        smallIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(IntVector intVector, int value, int rowCount) {
+        intVector.setSafe(rowCount, value);
+        intVector.setValueCount(rowCount + 1);
+    }
+
+    private static  void updateVector(BigIntVector bigIntVector, int value, int rowCount) {
+        bigIntVector.setSafe(rowCount, value);
+        bigIntVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(DecimalVector decimalVector, BigDecimal value, int rowCount) {
+        decimalVector.setSafe(rowCount, value);
+        decimalVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(Float4Vector float4Vector, float value, int rowCount) {
+        float4Vector.setSafe(rowCount, value);
+        float4Vector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(Float8Vector float8Vector, double value, int rowCount) {
+        float8Vector.setSafe(rowCount, value);
+        float8Vector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(VarCharVector varcharVector, String value, int rowCount) {
+        if (value != null) {
+            varcharVector.setIndexDefined(rowCount);
+            varcharVector.setValueLengthSafe(rowCount, value.length());
+            varcharVector.setSafe(rowCount, value.getBytes(StandardCharsets.UTF_8), 0, value.length());
+            varcharVector.setValueCount(rowCount + 1);
+        }
+        // TODO: not sure how to handle null string value ???
+    }
+
+    private static void updateVector(DateMilliVector dateMilliVector, Date date, int rowCount) {
+        //TODO: Need to handle Timezone
+        dateMilliVector.setValueCount(rowCount + 1);
+        if (date != null) {
+            dateMilliVector.setSafe(rowCount, date.getTime());
+        } else {
+            dateMilliVector.setNull(rowCount);
+        }
+    }
+
+    private static void updateVector(TimeMilliVector timeMilliVector, Time time, int rowCount) {
+        timeMilliVector.setValueCount(rowCount + 1);
+        if (time != null) {
+            timeMilliVector.setSafe(rowCount, (int) time.getTime());
+        } else {
+            timeMilliVector.setNull(rowCount);
+        }
+    }
+
+    private static void updateVector(TimeStampVector timeStampVector, Timestamp timestamp, int rowCount) {
+        //TODO Need to handle timezone ???
```

Comment:
After researching this, it seems that there's no guidance on the spec, and each driver handles this a bit differently. The only sane thing to do is assume that this represents a local timestamp and get the number of seconds as is...